### ISSUE TITLE
Blocks description is misleading

### ISSUE THREAD

**Issue by togakangaroo** [DrkSephy/es6-cheatsheet#84]
Blocks have been in js for a long time - they’re the reason

```
return
{
   foo: 123
}
```

breaks since it is interpreted as 

```
return undefined;
{   //block
   foo:  //label
      123; //expression
}
```

That being said, blocks haven’t been very useful before due to lack of block scope, but that’s not different than the earlier discussion of `let` and `const`

Blocks don’t really do the same thing as IIFEs in enough situations to call then replacements I think. They will not limit `var` declarations, but *more importantly* they won't limit `function` declarations which will be hoisted into a higher scope

**Comment by DrkSephy** [comment]
You're absolutely right. Blocks don't cover all the situations to replace IIFEs, but they should be considered first. Maybe I can update the example to better reflect this - I'll think about a good way to phrase it.

**Comment by Zbigbear93** [comment]
Where are the latest comments
